ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -327,6 +328,27 @@ calcParserModules ghcFlavor = do
         [ if ghcFlavor `elem` [ GhcMaster, Ghc8101 ] then "GHC.Hs.Dump" else "HsDumpAst"]
   return $ nubSort (modules ++ extraModules)
 
+-- Avoid duplicate symbols with HSghc-heap (see issue
+-- https://github.com/digital-asset/ghc-lib/issues/210).
+applyPatchHeapClosures :: GhcFlavor -> IO ()
+applyPatchHeapClosures _ = do
+  writeFile "libraries/ghc-heap/cbits/HeapPrim.cmm" .
+    replace
+      "aToWordzh"
+      "Ghclib_aToWordzh" .
+    replace
+      "reallyUnsafePtrEqualityUpToTag"
+      "Ghclib_reallyUnsafePtrEqualityUpToTag"
+    =<< readFile' "libraries/ghc-heap/cbits/HeapPrim.cmm"
+  writeFile "libraries/ghc-heap/GHC/Exts/Heap/Closures.hs" .
+    replace
+      "\"aToWordzh\""
+      "\"Ghclib_aToWordzh\"" .
+    replace
+      "\"reallyUnsafePtrEqualityUpToTag\""
```

Comment:
Do we need the quotes here? If it really was ambiguous without the quotes the other replacement is also dangerous so I’m not sure it helps much.

Reply:
The phrasing makes sure `reallyUnsafePtrEqualityUpToTag` in Closures.hs is replaced but not `reallyUnsafePtrEqualityUpToTag#`. I didn't test not doing it that way (expecting it to fail) and put my focus instead on carefully checking what had to happen and verifying the result. In HeapPrim.cmm there was no chance of the substitutions going wrong in that fashion and thus would be correct applied unconditionally.